src/psyclone/tests/psyir/frontend/fparser2_fortran_use_test.py

```diff
@@ -49,6 +49,33 @@
     DataSymbol, AutomaticInterface, INTEGER_SINGLE_TYPE
 
 
+def test_use_return(parser):
+    ''' Check the the Fparser correctly handles when a function uses a
+    return variable with a kind defined inside the function.'''
+    processor = Fparser2Reader()
+    reader = FortranStringReader("real(rkind) function x()\n"
+                                 "  use my_mod, only: rkind\n"
+                                 "  x = 1.0_rkind\n"
+                                 "end function x\n")
+    ptree = parser(reader)
+    processor.generate_psyir(ptree)
+
+
+def test_use_return2(parser):
+    ''' Check the the Fparser correctly handles when a function uses a
+    return variable with a kind defined inside the parent module.'''
+    processor = Fparser2Reader()
+    reader = FortranStringReader("module mymod\n"
+                                 "  use my_mod, only: rkind\n"
+                                 " contains\n"
+                                 "real(rkind) function x()\n"
+                                 "  x = 1.0_rkind\n"
+                                 "end function x\n"
+                                 "end module mymod")
+    ptree = parser(reader)
+    processor.generate_psyir(ptree)
```

Comment:
Again, please could you check the resulting symbol table. Also, please could you check when the function has an explicit visibility (e.g. `private`), just to make sure.

Reply:
Added an extra code input to do this explicitly.